dockers-install.sh

```diff
@@ -0,0 +1,72 @@
+# Update
+sudo apt update
+
+# Installing ca-certificates
+if (test -d ca-certificates)
+then
+  echo "ca-certificates already installed"
+else
+  echo "Installing ca-certificates"
+  sudo apt install -y ca-certificates
+fi
+
+# Installing curl
+if (which curl)
+then
+  echo "curl already installed"
+else
+  echo "Installing curl"
+  sudo apt install -y curl 
+fi
+
+# Creating a keyrings directory 
+if (test -d /etc/bin/keyrings)
+then
+  echo "creating keyrings"
+else
+  echo "Creating keyrings"
+  sudo install -m 0755 -d /etc/apt/keyrings
+fi
+
+# Adding a gpg keyring
+if (stat -c "%a" /etc/bin/keyrings/docker.asc)
+then
+  echo "gpg keyring already exists"
+else
+  echo "Installing gpg keyring"
+  sudo curl -fsSL https://download.docker.com/linux/debian/gpg -o /etc/apt/keyrings/docker.asc
+fi
+ 
+# setting permissions 
+if (test -f /etc/bin/keyrings/docker.gpg)
+then
+  echo " gpg keyring already configured"
+else
+  echo "Setting permissions for gpg keyring"
+  sudo chmod a+r /etc/apt/keyrings/docker.gpg
+fi
+
+# Intializing repository, and placing copies in .list files
+if (stat -c "%n" /etc/os-release && echo "$VERSION_CODENAME")
+then 
+  echo "Repository already initialized"
+else
+  echo "Initializing repo"
+  echo \
+  "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.asc] https://download.docker.com/linux/debian \
+  $(. /etc/os-release && echo "$VERSION_CODENAME") stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
```

Comment:
what is `. /etc/os-release` doing?

Reply:
- It is a susbshell that starts with a path to the file that contains the Operating system type && echoing an env. variable which I believe is identifying the operating system release code name.